src/NHibernate/Linq/NestedSelects/NestedSelectRewriter.cs

```diff
@@ -16,7 +16,18 @@ namespace NHibernate.Linq.NestedSelects
 {
 	static class NestedSelectRewriter
 	{
-		private static readonly MethodInfo CastMethod = EnumerableHelper.GetMethod("Cast", new[] { typeof (IEnumerable) }, new[] { typeof (object[]) });
+		private static readonly MethodInfo CastMethod = ReflectionHelper.GetMethod(
```

Comment:
I would love these fields to be concentrated in `EnumerableHelper` class (as `internal static readonly` fields), or other class, if we are making `EnumerableHelper` obsolete.